.github/workflows/java-build.yml

```diff
@@ -0,0 +1,135 @@
+name: CI-java
+on:
+  pull_request:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+    outputs:
+      modules:
+        description: "Stream reactor collection of java modules"
+        value: ${{ jobs.initiate-java-modules.outputs.matrix }}
+
+jobs:
+
+  initiate-java-modules:
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.java-mods.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'gradle'
+      - name: Generate modules lists
+        run: cd 'java-connectors' && ./gradlew releaseModuleList
+        env:
+          JVM_OPTS: -Xmx512m
+      - name: Read java modules lists
+        id: java-mods
+        run: |
+          echo "::set-output name=matrix::$(cat ./java-connectors/gradle-modules.txt)"
```

Comment:
This line causes this warning:

> [initiate-java-modules](https://github.com/lensesio/stream-reactor/actions/runs/8700471462/job/23861491700#step:5:8)
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

It sounds daunting but actually it's really easy to fix, we cleaned it up on master branch.

Reply:
no worries, already addressed :) Thank you!